src/Microsoft.Health.Dicom.Core/Extensions/DicomElementExtensions.cs

```diff
@@ -0,0 +1,164 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.Validation;
+
+namespace Microsoft.Health.Dicom.Core.Extensions
+{
+    /// <summary>
+    /// Extension methods for <see cref="DicomElement"/>.
+    /// </summary>
+    public static class DicomElementExtensions
+    {
+        private static readonly IReadOnlyDictionary<DicomVR, Func<DicomElement, object>> Readers = new Dictionary<DicomVR, Func<DicomElement, object>>()
+        {
+            { DicomVR.AE, GetStringValue },
+            { DicomVR.AS, GetStringValue },
+            { DicomVR.AT, GetDicomTagAsULongValue },
+            { DicomVR.CS, GetStringValue },
+            { DicomVR.DA, GetDAValue },
+            { DicomVR.DS, GetStringValue },
+            { DicomVR.DT, GetDTValue },
+            { DicomVR.FL, GetFloatValue },
+            { DicomVR.FD, GetDoubleValue },
+            { DicomVR.IS, GetStringValue },
+            { DicomVR.LO, GetStringValue },
+            { DicomVR.PN, GetStringValue },
+            { DicomVR.SH, GetStringValue },
+            { DicomVR.SL, GetIntValue },
+            { DicomVR.SS, GetShortValue },
+            { DicomVR.TM, GetTMValue },
+            { DicomVR.UI, GetStringValue },
+            { DicomVR.UL, GetUIntValue },
+            { DicomVR.US, GetUShortValue },
+        };
+
+        /// <summary>
+        /// Get value of DicomElement.
+        /// </summary>
+        /// <remarks>VM of the DicomElement must be 1. Value rather than 1 will return null.</remarks>
+        /// <param name="element">The DicomElement</param>
+        /// <returns>The value.</returns>
+        public static object GetSingleValue(this DicomElement element)
+        {
+            EnsureArg.IsNotNull(element, nameof(element));
+            if (element.Count != 1)
+            {
+                return null;
+            }
+
+            if (!Readers.ContainsKey(element.ValueRepresentation))
```

Comment:
nit: use `TryGetValue` so you don't need to look up the func twice

Reply:
This class is removed.